File: apps/backend/src/rhesis/backend/app/schemas/prompt.py
from typing import Optional

from pydantic import UUID4

from rhesis.backend.app.schemas.base import Base


# Base Prompt Schema
class PromptBase(Base):
    content: str
    category_id: Optional[UUID4] = None
    topic_id: Optional[UUID4] = None
    language_code: str
    behavior_id: Optional[UUID4] = None
    expected_response: Optional[str] = None
    user_id: Optional[UUID4] = None
    organization_id: Optional[UUID4] = None
    status_id: Optional[UUID4] = None


# Prompt Create schema
class PromptCreate(PromptBase):
    pass


# Prompt Update schema - extending PromptBase
class PromptUpdate(PromptBase):
    content: Optional[str] = None  # Override to make 'text' optional for updates
    language_code: Optional[str] = None  # Override to make 'language_code' optional for updates


# Read schema (optional, if it contains extra fields)
class Prompt(PromptBase):
    pass

File: apps/frontend/src/utils/api-client/interfaces/prompt.ts
import { UUID } from 'crypto';

// Base prompt interface
export interface PromptBase {
  content: string;
  language_code: string;
  expected_response?: string;
}

// Create and update interfaces
export type PromptCreate = PromptBase;

export type PromptUpdate = Partial<PromptBase>;

// Full prompt entity with id and timestamps
export interface Prompt extends PromptBase {
  id: UUID;
}
